### PR TITLE
update file handling to account for writing in binary mode

### PR DESCRIPTION
**Pull request by briandowns**
Signed-off-by: Brian Downs <brian.downs@gmail.com>

### Well detailed description of the change :

Adds the ability to write in binary mode.

With this change, the following code works:

```cs
#!/usr/bin/env dictu

import HTTP;
import System;

{ // main
    const opts = {
        "timeout": 20,
        "headers": [
            "User-Agent: Dictu"
        ],
        "insecure": true,
        "follow-redirects": true
    };
        
    const client = HTTP.newClient(opts);

    const res = client.get("https://curl.se/download/curl-7.86.0.zip");
    const result = res.unwrap();

    with("curl-7.86.0.zip", "wb") {
        file.write(result.content);
    }

    System.exit(0);
}
```

Resolves: #

#567 

### Type of change:

- [ ] Bug fix

- [x] New feature

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#

### Housekeeping

- [ ] Tests have been updated to reflect the changes done within this PR (if applicable).

- [x] Documentation has been updated to reflect the changes done within this PR (if applicable).

#

### CODE REVIEW

File: src/vm/datatypes/files.c [hunk 1]
```diff
@@ -15,13 +15,19 @@ static Value writeFile(DictuVM *vm, int argCount, Value *args) {
...
 
-    int charsWrote = fprintf(file->file, "%s", string->chars);
-    fflush(file->file);
+    int charsWrote = 0;
+
+    if (strcmp(file->openType, "wb") == 0) {
+        charsWrote = fwrite(string->chars, 1, string->length, file->file);
+    } else {
+        charsWrote = fprintf(file->file, "%s", string->chars);
+        fflush(file->file);
```
Comment (Jason2605) [inline src/vm/datatypes/files.c:29]: Probably worth moving the flush call outside the if blocks so it's called regardless

File: src/vm/datatypes/files.c [hunk 2]
```diff
@@ -82,7 +88,10 @@ static Value readFullFile(DictuVM *vm, int argCount, Value *args) {
         buffer = SHRINK_ARRAY(vm, buffer, char, fileSize + 1, bytesRead + 1);
     }
 
-    buffer[bytesRead] = '\0';
+    if (strcmp(file->openType, "rb") != 0) {
+        buffer[bytesRead] = '\0';
+    }
```
Comment (Jason2605) [inline src/vm/datatypes/files.c:93]: We will want to null terminate here as Dictu expects all strings to be null terminated. Is there a reason you didn't, did you find something break or?

**Comment by briandowns** [comment]
I clicked on the "request review" button only to let you know the changes are in. No rush at all. I saw in another PR you said you have a busy week so figured I'd drop this note so not to seem pushy. :D

**Comment by Jason2605** [comment]
Haha I would prefer to get the extra emails than miss something, so cheers for the reminder! Yeah busy one this week, will try to get round to it

**Comment by Jason2605** [comment]
Thanks for this

**Pull request merged**